apps/mysql.go

```diff
@@ -15,13 +15,75 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
+	"os"
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,hostname_port|file"`
+
+	Password  string `yaml:"password" validate:"omitempty"`
+	Username  string `yaml:"username" validate:"omitempty"`
+	Transport string `yaml:"transport" validate:"omitempty"`
+}
+
+const defaultMySQLTCPEndpoint = "localhost:3306"
+const defaultMySQLUnixEndpoint = "/var/run/mysqld/mysql.sock"
+
+func (r MetricsReceiverMySQL) Type() string {
+	return "mysql"
+}
+
+func (r MetricsReceiverMySQL) Pipelines() []otel.Pipeline {
+	if r.Endpoint == "" {
+		if _, err := os.Stat(defaultMySQLUnixEndpoint); err != nil {
+			r.Transport = "unix"
+			r.Endpoint = defaultMySQLUnixEndpoint
+		} else {
+			r.Endpoint = defaultMySQLTCPEndpoint
+		}
+	}
+
+	return []otel.Pipeline{{
+		Receiver: otel.Component{
+			Type: "mysql",
+			Config: map[string]interface{}{
+				"collection_interval": r.CollectionIntervalString(),
+				"endpoint":            r.Endpoint,
+				"username":            r.Username,
+				"password":            r.Password,
+				"transport":           r.Transport,
+			},
+		},
+		Processors: []otel.Component{
+			otel.NormalizeSums(),
+			otel.MetricsTransform(
+				otel.AddPrefix("workload.googleapis.com"),
+			),
+		},
+	}}
+}
+
+func init() {
+	confgenerator.MetricsReceiverTypes.RegisterType(func() confgenerator.Component { return &MetricsReceiverMySQL{} })
```

Comment:
Style nit: I would just use two init() functions and keep the other one at the end of the file.

Reply:
moved to two seperate funcs.